src/store/singing.ts

```diff
@@ -3449,6 +3461,121 @@ export const singingStore = createPartialStore<SingingStoreTypes>({
       return Math.max(1, lastNoteEndTime + 1);
     },
   },
+
+  EXPORT_SONG_PROJECT: {
+    action: createUILockAction(
+      async (
+        { state, getters, actions },
+        { fileType, fileTypeLabel },
+      ): Promise<SaveResultObject> => {
+        const fileBaseName = generateDefaultSongFileBaseName(
+          getters.PROJECT_NAME,
+          getters.SELECTED_TRACK,
+          getters.CHARACTER_INFO,
+        );
+        const project = ufProjectFromVoicevox(
+          {
+            tempos: state.tempos,
+            timeSignatures: state.timeSignatures,
+            tpqn: state.tpqn,
+            tracks: state.trackOrder.map((trackId) =>
+              getOrThrow(state.tracks, trackId),
+            ),
+          },
+          fileBaseName,
+        );
+
+        // 複数トラックかつ複数ファイルの形式はディレクトリに書き出す
+        if (
+          state.trackOrder.length > 1 &&
+          // NOTE: as MultiFileProjectFormat しないと型エラーが発生する
+          multiFileProjectFormats.includes(fileType as MultiFileProjectFormat)
+        ) {
+          const dirPath = await window.backend.showSaveDirectoryDialog({
+            title: "プロジェクトを書き出し",
+          });
+          if (!dirPath) {
+            return { result: "CANCELED", path: "" };
+          }
+
+          const extension = projectFileExtensions[fileType];
+          const tracksBytes = await ufProjectToMultiFile(
+            project,
+            fileType as MultiFileProjectFormat,
+          );
+
+          let firstFilePath = "";
+          for (const [i, trackBytes] of tracksBytes.entries()) {
+            const filePath = await actions.GENERATE_FILE_PATH_FOR_TRACK_EXPORT({
+              trackId: state.trackOrder[i],
+              directoryPath: dirPath,
+              extension,
+            });
+            if (i === 0) {
+              firstFilePath = filePath;
+            }
+
+            const result = await actions.EXPORT_FILE({
+              filePath,
+              content: trackBytes,
+            });
+            if (result.result !== "SUCCESS") {
+              return result;
+            }
+          }
+
+          return {
+            result: "SUCCESS",
+            path: firstFilePath,
+          };
```

Comment:
細かいけど、万が一`""`なパスが返ってしまったら結構危険な気がするので、初期値はundefinedのが良さそう。
```suggestion
          let firstFilePath;
          for (const [i, trackBytes] of tracksBytes.entries()) {
            const filePath = await actions.GENERATE_FILE_PATH_FOR_TRACK_EXPORT({
              trackId: state.trackOrder[i],
              directoryPath: dirPath,
              extension,
            });
            if (i === 0) {
              firstFilePath = filePath;
            }

            const result = await actions.EXPORT_FILE({
              filePath,
              content: trackBytes,
            });
            if (result.result !== "SUCCESS") {
              return result;
            }
          }
          if (firstFilePath == undefined) {
            throw new Error("firstFilePath is undefined.");
          }

          return { result: "SUCCESS", path: firstFilePath };
```